packages/core/src/animation/Animator.ts

```diff
@@ -0,0 +1,465 @@
+import { AnimatorState } from "./AnimatorState";
+import { InterpolableValueType } from "./enums/InterpolableValueType";
+import { AnimatorRecorderMode } from "./enums/AnimatorRecorderMode";
+import { InterpolableValue } from "./KeyFrame";
+import { AnimatorControllerLayer } from "./AnimatorControllerLayer";
+import { AnimatorController } from "./AnimatorController";
+import { Quaternion, Vector2, Vector3, Vector4 } from "@oasis-engine/math";
+import { Component } from "../Component";
+import { Entity } from "../Entity";
+import { AnimatorUtils } from "./AnimatorUtils";
+import { AnimateProperty } from "./enums/AnimateProperty";
+import { AnimatorLayerBlendingMode } from "./enums/AnimatorLayerBlendingMode";
+import { PlayType } from "./enums/PlayType";
+
+/**
+ * The controller of the animation system.
+ */
+export class Animator extends Component {
+  /**
+   * The mode of the Animator recorder.
+   */
+  recorderMode: AnimatorRecorderMode = AnimatorRecorderMode.Offline;
+  /**
+   * The playback speed of the Animator. 1 is normal playback speed.
+   */
+  speed: number = 1;
+
+  /**
+   * Get the AnimatorController that controls the Animator.
+   */
+  get animatorController(): AnimatorController {
+    return this._animatorController;
+  }
+
+  /**
+   * Set the AnimatorController that controls the Animator.
+   */
+  set animatorController(animatorController: AnimatorController) {
+    this._animatorController = animatorController;
+    if (!animatorController) return;
+    animatorController._setTarget(this.entity);
+  }
+
+  /**
+   * Get all layers from the AnimatorController which belongs this Animator .
+   */
+  get layers(): Readonly<AnimatorControllerLayer[]> {
+    return this._animatorController?.layers || [];
+  }
+
+  private _animatorController: AnimatorController;
+  private _diffValueFromBasePos: InterpolableValue;
+  private _diffFloatFromBasePos: number = 0;
+  private _diffVector2FromBasePos: Vector2 = new Vector2();
+  private _diffVector3FromBasePos: Vector3 = new Vector3();
+  private _diffVector4FromBasePos: Vector4 = new Vector4();
+  private _diffQuaternionFromBasePos: Quaternion = new Quaternion();
+  private _tempVector3: Vector3 = new Vector3();
+  private _tempQuaternion: Quaternion = new Quaternion();
+
+  /**
+   * @param entity - The entitiy which the animator component belongs to.
+   */
+  constructor(entity: Entity) {
+    super(entity);
+  }
+
+  /**
+   * Plays a state by name.
+   * @param stateName The state name.
+   * @param layerIndex The layer index(default 0).
+   * @param normalizedTime The time offset between 0 and 1(default 0).
+   */
+  playState(stateName: string, layerIndex: number = 0, normalizedTime: number = 0): void {
```

Comment:
normalizedTime is a litte hard to understand. 
If this param mean the offset of start play, we should call 'offsetTime' and write this time is normalized in commets

Reply:
done